### PR TITLE
spec for exam controller

### PR DESCRIPTION
**Pull request by ductvh-3178**
## Related Tickets
#51089

## WHAT (optional)
- write rspec for exams controller
98% lines covered
60 relevant lines. 56 lines covered and 2 lines missed.

## HOW
- add 11 test case for index, search,new,show, create, update action

## Evidence (Screenshot or Video)
Peek 2022-07-13 17-24

image

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -16,6 +17,7 @@ gem "net-smtp", require: false
 gem "pagy"
 gem "puma", "~> 5.0"
 gem "rails", "~> 6.1.6"
+gem "rails-controller-testing"
```
Comment (vanvtt-0952) [inline Gemfile:20]: Cái nì có cần dùng cho all môi trường ko hay chỉ cho test cu

File: app/controllers/exams_controller.rb [hunk 2]
```diff
@@ -73,9 +75,9 @@ def load_user
     @user = current_user
   end
 
-  def exam_params
-    params.require(:exam).permit(:user_id, :subject_id, :id)
-  end
+  # def exam_params
```
Comment (vanvtt-0952) [inline app/controllers/exams_controller.rb:78]: Sao comment đấy

File: app/controllers/exams_controller.rb [hunk 3]
```diff
@@ -90,10 +92,10 @@ def update_details answers
     end
   end
 
-  def user_not_correct
-    flash[:danger] = t ".user_not_correct"
-    redirect_to :root
-  end
+  # def user_not_correct
```
Comment (vanvtt-0952) [inline app/controllers/exams_controller.rb:95]: Ơ sao chú lani comment codethees

Comment (ductvh-3178) [reply app/controllers/exams_controller.rb:95]: 2 cái method đó không dùng á chị

**Pull request merged**